src/main/java/emiya/storage/Storage.java

```diff
@@ -0,0 +1,164 @@
+package emiya.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import emiya.emiyaexception.CreateDirectoryFailException;
+import emiya.emiyaexception.InvalidDateException;
+import emiya.emiyaexception.WrongDateFormatException;
+import emiya.task.Deadline;
+import emiya.task.Event;
+import emiya.task.Task;
+import emiya.task.TaskList;
+import emiya.task.ToDo;
+
+/**
+ * A class that contains methods that handle reading from a file and writing to a file.
+ */
+public class Storage {
+    public Storage() {
+
+    }
+
+    /**
+     * A method that reads a text file and returns the contents of the file as a String.
+     * @param fileName The name of the file that is to be accessed.
+     * @param dirName The name of the directory that contains the file to be accessed.
+     * @return A String containing the contents of the file indicated by fileName.
+     */
+    public String fileContents(String fileName, String dirName) {
+        String path = Paths.get("").toAbsolutePath().toString();
+        String pathToDataDir = Paths.get(path, dirName).toString();
+        Path pathToDataDoc = Paths.get(pathToDataDir, fileName);
+        String res = "";
+        try {
+            byte[] bytes = Files.readAllBytes(pathToDataDoc);
+            String content = new String(bytes);
+            res = content;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return res;
+    }
+
+    /**
+     * A method that creates a directory if it does not exist.
+     * @param dirName The name of the directory to be created if it does not already exist.
+     * @throws CreateDirectoryFailException When the directory cannot be created.
+     */
+    public void createDirectory(String dirName) throws CreateDirectoryFailException {
+        String path = Paths.get("").toAbsolutePath().toString();
+        String pathToDataDir = Paths.get(path, dirName).toString();
+        File dataDir = new File(pathToDataDir);
+
+        if (!dataDir.exists()) {
+            boolean result = dataDir.mkdirs();

```

Comment:
Instead of using general terms like result, use has, is or was for boolean variables.